tests/test_settings.py

```diff
@@ -666,3 +666,45 @@ def test_update_brand_logo(client: FlaskClient, admin: User) -> None:
     resp = client.get(logo_url, follow_redirects=True)
     # yes this check is ridiculous. why? because we redirect not-founds instead of actually 404-ing
     assert "That page doesn" in resp.text
+
+
+@pytest.mark.usefixtures("_authenticated_admin")
+def test_update_directory_intro_text(client: FlaskClient, admin: User) -> None:
+    """Test updating the directory intro text and ensure data integrity on failure."""
+
+    # Step 1: Set initial intro text and confirm it loads correctly
```

Comment:
It might be better to have this test check the endpoint when the value isn't set to we know that a missing/`null` value in the DB doesn't break things.